### ISSUE TITLE
Validation step fails if any NaN present

### ISSUE THREAD

**Issue by mmacander** [nasa-nccs-hpda/above-shrubs#8]
The validation step fails if there are any NaN present in the predictions. I am not sure where the NaN came from but there must have been some no data in the predictor stack? I worked around this by also exporting a CSV version of the predicted and label arrays that I could summarize in R.

(Added these two lines near the end of chm_pipeline.py:)
np.savetxt(os.path.join(self.metadata_dir, 'test_predictions.csv'), predictions.ravel(), delimiter=',', header='pred', comments='')
np.savetxt(os.path.join(self.metadata_dir, 'test_labels.csv'), test_labels.ravel(), delimiter=',', header='test', comments='')

These files are huge though, 7.4 GB each for ~4800 tiles. Can be improved by specifying a numeric data format but initial attempt failed on NaN conversion.

**Comment by jordancaraballo** [comment]
Drop some of the no-data stuff, get a metric of the data that was dropped "10% of the data was dropped because of no-data"

**Comment by jordancaraballo** [comment]
Synthetic no-data generation:

```python
import random
if random.choice([True, False]):
    test_data_tile[255, 255, 0] = np.nan
```

Fixed in the new commit from 0.2.0. Promoting changes to production in the next merge request.
